src/pages/smart-contracts/public-registry-tutorial.md

```diff
@@ -0,0 +1,612 @@
+In addition to `tx_result`, the transaction object also contains a timestamp (`burn_block_time_iso`) and a `contract_call` object. The contract call object has properties like the function name (`contract_call.function_name`). Using this, the final representation for `register` transactions should contain the deserialized registry id and the timestamp. It could look like this:
+
+```js
+{
+  activities.map((activity, key) => {
+    if (activity.contract_call.function_name === 'register') {
+      const result = hexToCV(activity.tx_result.hex.substr(2));
+      return (
+        <React.Fragment key={key}>
+          Entry {result.value.toString()} was registered at {activity.burn_block_time_iso}.{' '}
+        </React.Fragment>
+      );
+    } else {
+      return null;
+    }
+  });
+}
+```
+
+-> Note: The `AccountsApi` and other Api methods provide parameters to page through the results using `limit` and `offset`. See [the docs](https://blockstack.github.io/stacks-blockchain-api/client/interfaces/getaccounttransactionsrequest.html) for more details.
+
+=> Congratulations! You just implemented a list of recent activities that was fetched from the blockchain.
+
+![How recent activities look like](/images/todos-recent-activities.png)
+
+## Fetch the first to-do list
+
+There are two other ways to get state information from the blockchain: read-only functions and data map entries. Read-only functions were already discussed in the [Clarity counter tutorial](/smart-contracts/counter-tutorial). They do not require a transaction to complete. Data maps in Clarity are maps that can be read by any user. See the [Clarity reference](https://docs.blockstack.org/references/language-clarity#define-map) for more details.
+
+The `todo-registry` contract defines a read-only function `owner-of?` that returns the owner of a registry entry and a data map for details about entries:
+
+```clarity
+(define-read-only (owner-of (registration-id uint)) ... )
+
+(define-map registry
+  ((registry-id uint))
+  (
+    (name (buff 30))
+    (url (buff 255))
+  )
+)
+```
+
+Let's add the owner information and the details for the first ever registered to-do list (with `registry-id` 1) to the `RecentActivities` component. The `SmartContractsApi` of the client library provides methods to read these data from the blockchain.
+
+### Step 1: Add state variable for first registration
+
+At first, add a new state variable `firstRegistration`.
+
+```js
+// src/components/RecentActvities.jsx
+
+const [firstRegistration, setFirstRegistration] = useState();
+```
+
+Then, define a new callback `fetchRegistration` that will contain the calls to the blockchain and update the state variable. Add the following code after the `fetchRegistration` callback and update the effect hook.
+
+```js
+const fetchRegistration = useCallback(async () => {
+  // fetch newest registration
+  // fetch owner
+  // fetch public URL and name
+});
+
+useEffect(() => {
+  fetchActivities();
+  fetchRegistration();
```

Comment:
```suggestion
 Promise.all([fetchActivities(), fetchRegistration()];
```

Reply:
@aulneau What does the `void` do in front of `Promise.all`?

Reply:
@friedger https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-floating-promises.md#ignorevoid

> This allows you to stop the rule reporting promises consumed with void operator. This can be a good way to explicitly mark a promise as intentionally not awaited.